### DISCUSSION TITLE
pageViewMode="book"

### DISCUSSION THREAD

**Discussion by Max-Morrelli** [stephanrauh/ngx-extended-pdf-viewer#947] [Q&A]
Hi, I'd like to compliment you on the library. I wanted to ask you for an info, in book view mode the second page never loads. Let me explain when I click to turn the page, the page on the right is displayed and not the one on the left. What am I doing wrong?

**Comment by stephanrauh** [comment]
It's not your fault. The bug you've discovered the reason why the book mode is still an experimental feature.

I suppose there's a workaround. When you use the chevrons to go to the next page, the second page is rendered.
![image](https://user-images.githubusercontent.com/3045767/134721226-d55764f9-f891-4787-ba16-c30e9b50f5be.png)

So probably you can implement a `(pageRendered)` listener turning to the un-rendered page. I'll try this tomorrow. If this approach works, I can implement a low-level solution to make the pages render without further ado.

However, I don't understand why my previous attempts didn't work. I suspect the book mode plugin causes so much CPU load that pdf.js stops working properly. I hope to be able to reduce the CPU load, but to do so, I need to understand how https://nodlik.github.io/StPageFlip/ works internally. In theory, it's easy - don't update the CSS rules if nothing has changed - but when I tried to convert the idea to code, things quickly become messy.

**Reply by Max-Morrelli** [reply]
Tnx for the answer. I wait for your feedback

**Reply by stephanrauh** [reply]
Remind me if you don't hear from me during the next couple of days. Today I didn't find time for maintaining the library. I should stop making promises. :)

**Reply by stephanrauh** [reply]
Please update to version 10.0.1. It uses the work-around I've been proposing. It's still not perfect, but most of the time, it does the trick now.

**Reply by Max-Morrelli** [reply]
thank you very much, I will update you as I implement it
